cava/cava/Cava2SystemVerilog.hs

```diff
@@ -114,14 +110,17 @@ computeVectors instances
 
 computeVectors' :: Netlist.Primitive -> State NetlistGenerationState
                                         String
-computeVectors' (Netlist.UnsignedAdd m n a b s)
+computeVectors' (Netlist.BindPrimitive _ _ (Netlist.UnsignedAdd _ _) any_ab any_s)
   = do NetlistGenerationState v vDecs <- get
        let vA = VectorDeclaration AssignTo   v       HighToLow a
            vB = VectorDeclaration AssignTo   (v + 1) HighToLow b
            vC = VectorDeclaration AssignFrom (v + 2) HighToLow s
        put (NetlistGenerationState (v + 3) (vA:vB:vC:vDecs))
        return ("  assign v" ++ show (v + 2) ++ " = v" ++ show v ++
-               " + v" ++ show (v + 1) ++ ";")                                       
+               " + v" ++ show (v + 1) ++ ";")
+  where
+   (a, b) = Netlist.unsafeCoerce any_ab
```

Comment:
There are a number of manual coercions in this file im not happy with, I think I should write these functions in Coq so that any coercions needed are performed by Coq. Do you have any thoughts on this?